Dockerfile

```diff
@@ -1,36 +1,145 @@
+            HAVE_RULES="yes" \
+            CXXFLAGS="-O2 -DNDEBUG -Wall -Wno-sign-compare -Wno-unused-function -Wno-deprecated-declarations" \
+    # Compile RATS (and expat)
+    && tar -xvzf expat-2.0.1.tar.gz \
+    && cd expat-2.0.1 \
+    && ./configure \
+    && make \
+    && make install \
+    && cd .. \
+    && tar -xzvf rats-2.4.tgz \
+    && cd rats-2.4 \
+    && ./configure --with-expat-lib=/usr/local/lib \
+    && make \
+    && make install \
+    && ./rats \
+    && cd ..
+
+################################################################################
+
+# Final image based on the official sonar-scanner image
+FROM debian:10.5-slim
 
 LABEL maintainer="CATLab <catlab@cnes.fr>"
 
-# Add and set up Python tools
-# * pylint
-# * CNES pylint extension
-# * CNES pylintrc A_B, C, D
-COPY --chown=scanner-cli:scanner-cli pylintrc.d/ /opt/python/
+# Set variables for the sonar-scanner
+ENV SRC_DIR=/usr/src \
+    SONAR_SCANNER_HOME=/opt/sonar-scanner \
+    SONAR_USER_HOME=/opt/sonar-scanner/.sonar
+
+# Same workdir as the offical sonar-scanner image
+WORKDIR ${SRC_DIR}
+
+# Add an unprivileged user
+RUN addgroup sonar-scanner \
+    && adduser \
+            --home "$SONAR_SCANNER_HOME" \
+            --ingroup sonar-scanner \
+            --disabled-password \
+            --gecos "" \
+            sonar-scanner \
+    && mkdir -p "$SONAR_SCANNER_HOME/bin" \
+            "$SONAR_SCANNER_HOME/lib" \
+            "$SONAR_SCANNER_HOME/conf" \
+            "$SONAR_SCANNER_HOME/.sonar/cache" \
+            "$SONAR_SCANNER_HOME/.pylint.d" \
+    && chown -R sonar-scanner:sonar-scanner \
+            "$SONAR_SCANNER_HOME" \
+            "$SONAR_SCANNER_HOME/.sonar" \
+            "$SONAR_SCANNER_HOME/.pylint.d" \
+            "$SRC_DIR" \
+    && chmod -R 777 \
+            "$SONAR_SCANNER_HOME/.sonar" \
+            "$SONAR_SCANNER_HOME/.pylint.d" \
+            "$SRC_DIR"
+
+# Add sonar-scanner from builder
+COPY --from=builder /sonar-scanner/bin/sonar-scanner \
+    "$SONAR_SCANNER_HOME/bin"
+COPY --from=builder /sonar-scanner/lib \
+    "$SONAR_SCANNER_HOME/lib"
+# and our default sonar-scanner.properties
+COPY conf/sonar-scanner.properties "$SONAR_SCANNER_HOME/conf"
 
+# Add CppCheck from builder stage
+COPY --from=builder /usr/share/cppcheck /usr/share/cppcheck
+COPY --from=builder /usr/bin/cppcheck /usr/bin
+COPY --from=builder /usr/bin/cppcheck-htmlreport /usr/bin
+
+# Add RATS from builder stage
+COPY --from=builder /usr/local /usr/local
+
+# Add CNES pylintrc A_B, C, D
+COPY pylintrc.d/ /opt/python/
+
+# Download CNES pylint extension
 ADD https://github.com/cnescatlab/cnes-pylint-extension/archive/v5.0.0.tar.gz \
     /tmp/python/
 
-RUN apk add --no-cache --virtual .pylint-build-deps \
-        gcc=9.3.0-r0 \
-        python3-dev=3.8.2-r1 \
-        musl-dev=1.1.24-r2 \
+# Install tools
+RUN echo 'deb http://ftp.fr.debian.org/debian/ bullseye main contrib non-free' >> /etc/apt/sources.list \
```

Comment:
Codacy found an issue: [Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`](https://app.codacy.com/gh/cnescatlab/sonar-scanner/pullRequest?prid=6081579)